lib/matplotlib/dviread.py

```diff
@@ -828,14 +828,19 @@ def __new__(cls, filename):
         # store the unparsed lines (keyed by the first word, which is the
         # texname) and parse them on-demand.
         with open(filename, 'rb') as file:
-            self._unparsed = {line.split(b' ', 1)[0]: line for line in file}
+            self._unparsed = defaultdict(list)
+            for line in file:
+                tfmname = line.split(b' ', 1)[0]
+                self._unparsed[tfmname].append(line)
```

Comment:
I usually write this as
```
self._unparsed = {}
for line in file: ...; self._unparsed.setdefault(tfmname, []).append(line)
```
(defaultdict's autovivification always makes me a bit nervous) but I guess you should time whichever is fastest.

Reply:
Using `timeit.timeit('dviread.PsFontsMap(".../pdftex.map")', setup='from matplotlib import dviread')` taken 5 times, throwing away slowest and fastest times, the average for `defaultdict` is 0.202 microseconds, and for `setdefault` is 0.183 microseconds, for a map file that is 40827 lines long. Not sure if that's long or short.

Reply:
I'd say that's on the long side, thanks for checking.